component/azstorage/block_blob.go

```diff
@@ -895,6 +974,50 @@ func (bb *BlockBlob) stageAndCommitModifiedBlocks(name string, data []byte, inde
 	return nil
 }
 
+func (bb *BlockBlob) StageAndCommit(name string, bol *common.BlockOffsetList) error {
+	blobURL := bb.Container.NewBlockBlobURL(filepath.Join(bb.Config.prefixPath, name))
```

Comment:
Whats the difference between this and the method above?

Reply:
the one above is intended for write streams (w/o caching) where we need to have a window on the data buffer breaking it into blocks. I can look into cleaning it up further.